app/models/manageiq/providers/container_manager.rb

```diff
@@ -87,11 +87,17 @@ def port_show
       port.to_s
     end
 
+    # the purpose of this is to call:
+    #   - monitoring_manager.create_unless_exists
+    #   - infra_manager.build_unless_exists
     def endpoint_created(role)
       monitoring_endpoint_created(role) if respond_to?(:monitoring_endpoint_created)
```

Comment:
Hopefully not for much longer... https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/507

Reply:
I think we are doing this instead of a standard `:dependent => :destroy` b/c our current provider destroy is too resource dependent

Reply:
No I think this is because these child managers are tied to endpoints so you can delete the virtualization endpoint and we need to delete the child infra_manager